frontend/src/components/core/Sidebar/ThemedSidebar.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+import PageLayoutContext from "components/core/PageLayoutContext"
+import ThemeProvider from "components/core/ThemeProvider"
+import { createTheme, ThemeConfig } from "theme"
+import Sidebar, { SidebarProps } from "./Sidebar"
+
+const createSidebarTheme = (theme: ThemeConfig): ThemeConfig =>
+  createTheme(
+    {
+      sidebar: theme.emotion.colors.bgColor,
+      main: theme.emotion.colors.secondaryBg,
+      name: "Sidebar",
+    },
+    theme
+  )
+
+const ThemedSidebar = ({
+  theme,
+  children,
+  ...props
+}: Partial<SidebarProps>): ReactElement => {
+  const { activeTheme } = React.useContext(PageLayoutContext)
```

Comment:
I think you should be able to use emotion's `withTheme` higher order component here to get the theme passed in as a prop instead of plumbing `activeTheme` around manually using `PageLayoutContext`

Reply:
Unfortunately emotion's `withTheme` only provides us with the emotion theme whereas `activeTheme` is the entire theme config object with emotion theme, baseweb theme, name, etc. I was quite sad upon realizing this :'(